From b7c021f12a9502459662ce11210937fac136381c Mon Sep 17 00:00:00 2001
From: Yong Zhang <yong.zhang0@gmail.com>
Date: Thu, 13 Oct 2011 15:52:30 +0800
Subject: [PATCH 136/286] hrtimer: Add missing debug_activate() aid [Was: Re:
 [ANNOUNCE] 3.0.6-rt17]

On Fri, Oct 07, 2011 at 10:25:25AM -0700, Fernando Lopez-Lezcano wrote:
> On 10/06/2011 06:15 PM, Thomas Gleixner wrote:
> >Dear RT Folks,
> >
> >I'm pleased to announce the 3.0.6-rt17 release.
>
> Hi and thanks again. So far this one is not hanging which is very
> good news. But I still see the hrtimer_fixup_activate warnings I
> reported for rt16...

Hi Fernando,

I think below patch will smooth your concern?

Thanks,
Yong
---
 kernel/hrtimer.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/kernel/hrtimer.c b/kernel/hrtimer.c
index 27a3192..34b3232 100644
--- a/kernel/hrtimer.c
+++ b/kernel/hrtimer.c
@@ -1052,6 +1052,7 @@ int __hrtimer_start_range_ns(struct hrtimer *timer, ktime_t tim,
 			 * remove it again and report a failure. This avoids
 			 * stale base->first entries.
 			 */
+			debug_deactivate(timer);
 			__remove_hrtimer(timer, new_base,
 					timer->state & HRTIMER_STATE_CALLBACK, 0);
 		}
